src/GafferSceneUI/LightTool.cpp

```diff
@@ -0,0 +1,1266 @@
+	)
+	{
+		m_selectionDirty = true;
+		if( !m_dragging )
+		{
+			selectionChangedSignal()( *this );
+		}
+		m_handleInspectionsDirty = true;
+		m_priorityPathsDirty = true;
+	}
+
+	if( plug == activePlug() )
+	{
+		if( activePlug()->getValue() )
+		{
+			m_preRenderConnection = view()->viewportGadget()->preRenderSignal().connect(
+				boost::bind( &LightTool::preRender, this )
+			);
+		}
+		else
+		{
+			m_preRenderConnection.disconnect();
+			m_handles->setVisible( false );
+		}
+	}
+
+	if( plug == scenePlug()->transformPlug() )
+	{
+		m_handleTransformsDirty = true;
+	}
+
+	/// \todo Checking for dirty attributes overlaps with the job of the inspector
+	/// dirtied plug from `updateHandleInspections()`. Should we remove handling
+	/// inspector dirtied signals? The `gl:visualiser:scale` attribute is used to
+	/// place the handles, so we at least need to catch changes to that attribute.
+	if( plug == scenePlug()->attributesPlug() )
+	{
+		m_handleInspectionsDirty = true;
+		m_handleTransformsDirty = true;
+	}
+}
+
+void LightTool::preRender()
+{
+	if( !m_dragging )
+	{
+		updateSelection();
+		if( m_priorityPathsDirty )
+		{
+			m_priorityPathsDirty = false;
+			SceneGadget *sceneGadget = static_cast<SceneGadget *>( view()->viewportGadget()->getPrimaryChild() );
+			if( !selection().isEmpty() )
+			{
+				sceneGadget->setPriorityPaths( ContextAlgo::getSelectedPaths( view()->getContext() ) );
+			}
+			else
+			{
+				sceneGadget->setPriorityPaths( IECore::PathMatcher() );
+			}
+		}
+	}
+
+	if( m_handleInspectionsDirty )
+	{
+		updateHandleInspections();
+		m_handleInspectionsDirty = false;
+
+		for( auto &c : m_handles->children() )
+		{
+			auto handle = runTimeCast<LightToolHandle>( c );
+			if( handle->getVisible() )
+			{
+				m_handles->setVisible( true );
+				break;
+			}
+		}
+	}
+
+	if( m_handleTransformsDirty )
+	{
+		updateHandleTransforms( 75 );
+		m_handleTransformsDirty = false;
+	}
+}
+
+void LightTool::updateSelection() const
+{
+	if( !m_selectionDirty )
+	{
+		return;
+	}
+
+	if( m_dragging )
+	{
+		// In theory, an expression or some such could change the effective
+		// transform plug while we're dragging (for instance, by driving the
+		// enabled status of a downstream transform using the translate value
+		// we're editing). But we ignore that on the grounds that it's unlikely,
+		// and also that it would be very confusing for the selection to be
+		// changed mid-drag.
+		return;
+	}
+
+	m_selection.clear();
+	m_selectionDirty = false;
+
+	if( !activePlug()->getValue() )
+	{
+		return;
+	}
+
+	// If there's no input scene, then there's no need to
+	// do anything. Our `scenePlug()` receives its input
+	// from the View's input, but that doesn't count.
+	auto scene = scenePlug()->getInput<ScenePlug>();
+	scene = scene ? scene->getInput<ScenePlug>() : scene;
+	if( !scene )
+	{
+		return;
+	}
+
+	m_selection = ContextAlgo::getSelectedPaths( view()->getContext() );
+}
```

Comment:
This has its origins in TransformTool, where updating the selection had significant overhead (because really it was doing inspections). Since all we do here is retrieve a pointer from the context, I'm not sure what `m_selectionDirty` is achieving - perhaps this could be simpler?